leader/leader.go

```diff
@@ -0,0 +1,35 @@
+package leader
+
+// Status indicates leadership status
+type Status int
+
+const (
+	// StatusNotLeader means the current node is not a leader
+	StatusNotLeader Status = iota
+
+	// StatusLeader means the current node / instance is a leader
+	StatusLeader
+
+	// StatusUnknown indicates some exception happened while determining leadership.  Consumer will interpret accordingly.
+	StatusUnknown
+)
+
+// CheckLeaderFunc is all that a special backend needs to implement.  It can be used with the
+// NewPoller function to return a polling implementation of the Detector interface.
```

Comment:
This doc tells more about integration than what the function is.  If i were looking to integrate, i'd want to know what true/false mean, and when i should return an `error` (i.e. what happens if i return error?)